### PR TITLE
[8.0] Backport of v9 partner_identification module

### PR DESCRIPTION
**Pull request by rousseldenis**

### CODE REVIEW

**Comment by rousseldenis** [comment]
@lmignon Don't hesitate to review ;-)

**Comment by lmignon** [comment]
:+1:  (tested on runbot)

File: partner_identification/models/res_partner_id_category.py [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+#             http://www.camptocamp.at
+# © 2015 Antiun Ingenieria, SL (Madrid, Spain)
+#        http://www.antiun.com
+#        Antonio Espinosa <antonioea@antiun.com>
+# ©  2016 ACSONE SA/NV (<http://acsone.eu>)
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+
+
+from openerp import api, models, fields
+from openerp.exceptions import ValidationError, Warning
```
Comment (andhit-r) [inline partner_identification/models/res_partner_id_category.py]: Warning as UserError (according to newest OCA CONTRIBUTING.md)

File: partner_identification/models/res_partner_id_category.py [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+        """
+        self.ensure_one()
+        eval_context = self._validation_eval_context(id_number)
+        try:
+            safe_eval(self.validation_code,
+                      eval_context,
+                      mode='exec',
+                      nocopy=True)
+        except Exception as e:
+            raise Warning(
```
Comment (andhit-r) [inline partner_identification/models/res_partner_id_category.py]: raise UserError

**Review by andhit-r — CHANGES_REQUESTED** [review]
Functionality: Ok (tested locally)
Code: Few comments (see inline comments)

**Comment by rousseldenis** [comment]
@andhit-r Done!

**Review by andhit-r — APPROVED** [review]
:+1:

**Comment by andhit-r** [comment]
@mikevhe18 @azmimr67 @m-sani @raditv would you kindly review this PR

**Comment by mikevhe18** [comment]
:+1:  (Functionality tested on runbot)

**Pull request merged**